### PR TITLE
Docs: add introduction and guide to readme

### PR DESCRIPTION
**Pull request by lolmaus**
My attempt at explaining `scenario-tester`:

* starts with a definition of a problem;
* then explains the solution in high-level phrasing;
* provides code samples for basic usage with `fromProject`, `expand` and `forEachScenario`, letting developers quickly grasp the idea;
* explains `fixturify-project` usage for creating codebases from disk and JSON, adding dependencies to them;
* explains how to add different versions of the same dependency: `npm:some-npm-package--legacy@1.2.3`;
* explains reusing an customizing scenarios with `skip`, `only` and `map`;
* documents CLI: `list` and `output`.

[Rendered](https://github.com/ef4/scenario-tester/blob/9b69ec39ff04a1d5a53662035c47e649550cbe13/README.md) (only the very first paragraph pre-exists).

Corrections and additions are very welcome!

***

Please note that this PR does *not* supersede @mansona's take on readme docs in #10. Chris describes a recommended setup, more elaborate and specific than my basic guide.

#10 can be merged together with this one. The only thing I would like to change there is the title: `Getting started` → `Recommended setup` or something like that.

### CODE REVIEW

File: README.md
Comment (mansona) [inline README.md:78]: I think we can link to one or the other github or npm? they link back to each other so it doesn't matter which one we link to here

File: README.md
Comment (mansona) [inline README.md:119]: if you want to show addDependency here use fake/mock package names. These dependencies that are added have nothing to do with npm, they are created as fixturify-project Projects with a minimal setup to be considered an npm package and then added to your project

see https://github.com/embroider-build/embroider/blob/main/tests/scenarios/compat-addon-import-test.ts#L10-L32 for inspiration

File: README.md
Comment (mansona) [inline README.md:145]: delete this section 👍

File: README.md
Comment (mansona) [inline README.md:195]: delete this line 👍 let's not document things that don't exist

File: README.md
Comment (mansona) [inline README.md:286]: Maybe add a section about CI setup to the guide? Maybe take from https://github.com/ef4/scenario-tester/pull/10/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R84

essentially `--matrix` is designed to communicate with GitHub and tell it what tests are available 👍

**Review by mansona — APPROVED** [review]
some minor changes but otherwise looking good 🎉 

a quick note: I know that @lolmaus mentioned that this PR doesn't supersede https://github.com/ef4/scenario-tester/pull/10 but we discussed it and we think it should for now. We can revisit https://github.com/ef4/scenario-tester/pull/10 after this is merged and maybe add it as a Tutorial section, or start a `docs` folder at that point, but that discussion shouldn't block this PR 👍

File: README.md
Comment (lolmaus) [reply README.md]: Done.

File: README.md
Comment (lolmaus) [reply README.md]: I have mentioned that this flag is used in Github CI test matrix config, but have not provided full CI documentation. I suggest that it goes into `docs/` along with your project setup guide.

**Review by mansona — APPROVED** [review]
LGTM

**Pull request merged**
